packages/contact-center/task/src/helper.ts

```diff
@@ -0,0 +1,142 @@
+import {useState, useEffect, useCallback, useRef} from 'react';
+import {TASK_EVENTS, UseTaskListProps, UseTaskProps} from './task.types';
+import {ITask} from '@webex/plugin-cc';
+
+// Hook for managing the task list
+export const useTaskList = (props: UseTaskListProps) => {
+  const {cc} = props;
+  const [taskList, setTaskList] = useState<ITask[]>([]);
+  const isEventRegistered = useRef(false); // Ensure the event is registered only once
+
+  const handleIncomingTask = useCallback((task: ITask) => {
+    setTaskList((prev) => {
+      // Prevent duplicate tasks
+      if (prev.some((t) => t.data.interaction.interactionId === task.data.interaction.interactionId)) {
+        return prev;
+      }
+      return [...prev, task];
+    });
+  }, []);
+
+  const handleTaskEnded = useCallback((taskId: string) => {
+    setTaskList((prev) => prev.filter((task) => task.data.interaction.interactionId !== taskId));
+  }, []);
+
+  const handleTaskMissed = useCallback((taskId: string) => {
+    setTaskList((prev) => prev.filter((task) => task.data.interaction.interactionId !== taskId));
+  }, []);
```

Comment:
I dont know if there should be anything in the dependency array herer, @Kesari3008  can you comment on this.